components/popup/src/vwc-popup-base.ts

```diff
@@ -98,23 +121,36 @@ export class VWCPopupBase extends LitElement {
 
 	override connectedCallback(): void {
 		super.connectedCallback();
-		document.addEventListener('scroll', this.updatePosition);
+		// Save the initial display state so that it can be restored when the positioning is complete
+		this.initialDisplayState = this.open;
+		this.hide();
+		this.anchorEl = this.getAnchorById();
+		this.anchorEl?.addEventListener('change', this.updatePosition);
+		window.addEventListener('scroll', this.updatePosition);
```

Comment:
why window instead of document?

Reply:
@yinonov I saw an example in the **Floating UI** [documentation](https://floating-ui.com/docs/computePosition).

![Screen Shot 2022-01-27 at 11 46 58](https://user-images.githubusercontent.com/10883919/151334050-f554ef01-c63d-4e24-912a-3abd94e04bff.png)